From a441e6a6f4bea3f89667c6caec630218890124d5 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Thu, 1 Jul 2021 02:33:27 +0300
Subject: [PATCH] u_ether: Disable error message for SIOCGMIIPHY ioctl
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Some services such as collectd ัะบ net-snmp periodically send
the SIOCGMIIPHY command to the ecm0 interface. This command
is not supported by the u_ether driver and as a result an error
message is repeatedly displayed:

    ether_ioctl: [ecm0] error: ioctl called for unsupported cmd[35143]

In fact, there is nothing terrible about this, so simply disable the
error message for the SIOCGMIIPHY command.

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 drivers/usb/gadget/function/u_ether.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/gadget/function/u_ether.c b/drivers/usb/gadget/function/u_ether.c
index 0b3bc8c19c3a..98fcb8008d8f 100644
--- a/drivers/usb/gadget/function/u_ether.c
+++ b/drivers/usb/gadget/function/u_ether.c
@@ -1635,6 +1635,10 @@ static int ether_ioctl(struct net_device *dev, struct ifreq *ifr, int cmd)
 		rc = rmnet_ioctl_extended(dev, ifr);
 		break;
 
+	case SIOCGMIIPHY:
+		rc = -EINVAL;
+		break;
+
 	default:
 		pr_err("[%s] error: ioctl called for unsupported cmd[%d]",
 			dev->name, cmd);
-- 
2.25.1

